eva-pipeline/src/test/java/embl/ebi/variation/eva/pipeline/jobs/VariantAnnotConfigurationBatchTest.java

```diff
@@ -0,0 +1,230 @@
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author Diego Poggioli
+ *
+ * Test for {@link VariantAnnotConfigurationBatch}
+ */
+@IntegrationTest
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = { VariantAnnotConfigurationBatch.class, AnnotationConfig.class})
+public class VariantAnnotConfigurationBatchTest {
+
+    @Autowired
+    private JobLauncherTestUtils jobLauncherTestUtils;
+
+    @Autowired
+    public VariantJobsArgs variantJobsArgs;
+
+    private static String dbName;
+    private static MongoClient mongoClient;
+    private BufferedReader reader;
+    private File vepInputFile;
+    private DBObjectToVariantAnnotationConverter converter;
+
+    @Test
+    public void validAnnotationJob () throws Exception {
+        variantJobsArgs.loadArgs();
+
+        insertVariantsWithoutAnnotations(dbName, dbName);
+
+        if(vepInputFile.exists())
+            vepInputFile.delete();
+
+        assertFalse(vepInputFile.exists());
+
+        makeVepAnnotationTmpGzipFile();
+
+        JobExecution jobExecution = jobLauncherTestUtils.launchJob();
+
+        assertEquals(ExitStatus.COMPLETED, jobExecution.getExitStatus());
+        assertEquals(BatchStatus.COMPLETED, jobExecution.getStatus());
+
+        //check list of variants without annotation output file
+        assertTrue(vepInputFile.exists());
+        assertEquals("20\t60343\t60343\tG/A\t+", readLine(vepInputFile));
+
+        //check that documents have the annotation
+        DBCursor cursor = collection(dbName, dbName).find();
+
+        int cnt=0;
+        int consequenceTypeCount = 0;
+        while (cursor.hasNext()) {
+            cnt++;
+            VariantAnnotation annot = converter.convertToDataModelType((DBObject)cursor.next().get("annot"));
+            TestCase.assertTrue(annot.getConsequenceTypes() != null);
+            consequenceTypeCount += annot.getConsequenceTypes().size();
+        }
+
+        assertTrue(cnt>0);
+        assertEquals(1, consequenceTypeCount);
+    }
+
+    /**
+     * This simulate VEP prediction
+     *
+     * @throws IOException
+     */
+    private void makeVepAnnotationTmpGzipFile() throws IOException {
+        String vepAnnotation =
+                "20_60343_G/A\t20:63351\tG\tENSG00000178591\tENST00000608838\tTranscript\tupstream_gene_variant\t-\t-\t-\t-\t-\trs181305519\tDISTANCE=4540;STRAND=1;SYMBOL=DEFB125;SYMBOL_SOURCE=HGNC;HGNC_ID=18105;BIOTYPE=processed_transcript;GMAF=G:0.0005;AFR_MAF=G:0.0020\n";
+
+        try(FileOutputStream output = new FileOutputStream(variantJobsArgs.getPipelineOptions().getString("vepOutput"))) {
+            try(Writer writer = new OutputStreamWriter(new GZIPOutputStream(output), "UTF-8")) {
+                writer.write(vepAnnotation);
+            }
+        }
+
+    }
+
+    private String readLine(File outputFile) throws IOException {
+        if (reader == null) {
+            reader = new BufferedReader(new FileReader(outputFile));
```

Comment:
Not in `@Before` because the file doesn't exist yet. I'm using now try with resources so I don't need to close the buffer. But still initialized in the method and not in `@Before`